.github/workflows/pr-go-unit-filtered.yml

```diff
@@ -9,6 +9,8 @@ on:
     branches: ["main"]
   pull_request:
     types: [opened, edited, synchronize]
+    paths:
+      - "**.go"
```

Comment:
I think this needs to be expanded:
 - Go module files e.g. `go.mod` and `go.sum`
 - We have fixtures (`cli/internal/lockfile/testdata` and `cli/fixtures`) that should trigger unit tests if they change

Reply:
Yeah I suppose we want to err on the side of too many paths versus too few. Should I just include the entire `cli` directory?

Reply:
Probably? If we end up needing to get more specific we should probably err on the side of moving things out of `cli` than trying to enumerate everything that can affect the unit tests.